src/test/java/uk/ac/ox/ctl/lti13/stateless/Lti13Step3Test.java

```diff
@@ -176,7 +175,7 @@ public void testStep3Empty() throws Exception {
     private OAuth2AuthorizationRequest.Builder createAuthRequest() {
         Map<String, Object> additionalParameters = new HashMap<>();
         additionalParameters.put(OAuth2ParameterNames.REGISTRATION_ID, "test");
-        return OAuth2AuthorizationRequest.implicit()
+        return OAuth2AuthorizationRequest.authorizationCode()
```

Comment:
This is wrong and we really should be doing an implicit grant.

Reply:
Why is this wrong? LTI 1.3 is working fine in brightspace with the authorizationCode

Reply:
Oh yeah, it works because the other bits of the code work with an authorization code flow. But what's actually being implemented is an implicit grant and it's really misleading to people reading the code.

https://www.imsglobal.org/spec/security/v1p0/#openid_connect_launch_flow

Reply:
Then we'd have to change this. `IMPLICIT` got removed from Spring Security without any direct replacement. It's either `AUTHORIZATION_CODE` OR `CLIENT_CREDENTIALS`.
https://docs.spring.io/spring-security/reference/5.8/migration/servlet/oauth2.html#_authorizationgranttype

And according tho the abstract of your source: 
```
This security framework has three basic patterns for adoption:

- Use of the OAuth 2.0 Client Credential Grant mechanism to secure web services between trusted systems (this MAY make use of JSON Web Tokens, JWT, for the access tokens);
- Use of the OAuth 2.0 Authorization Code Grant mechanism to secure web services between systems where there is no pre-established trust relationship (this MAY make use of JWT for the access tokens);
- Use of OpenID Connect with JWT-based message exchanges to secure browser-instigated exchanges between a tool and the launching platform.

All new 1EdTech specifications MUST review the security patterns defined herein and adopt those that are suitable. 1EdTech MAY revise specifications that have already been published to align with these security patterns.
```

You can also see it under `4.2 Using OAuth 2.0 Authorization Code Grant`.

So, I personally don't see the reason why we couldn't move to authorization code. It would also be our preferred method. On the other hand, I'm not an expert regarding OAuth or LTI

Reply:
The LTI login uses an implicit grant so it's rather confusing telling people to configure Spring Security for and authorization code grant. The reason authorization code is in the specification is for other parts of the LTi specification.